packages/itertools/iter.pony

```diff
@@ -792,12 +879,16 @@ class Iter[A] is Iterator[A]
     ## Example
 
     ```pony
-    Iter[I64]([as I64: 1; 2; 1; 1; 3; 4; 1].values())
+    Iter[USize]([as USize: 1; 2; 1; 1; 3; 4; 1].values())
         .unique()
     ```
     `1 2 3 4`
     """
-  */
+    let set: HashSet[A!, H] = HashSet[A!, H](this.count())
+    for v in _iter do
```

Comment:
This won't work for infinite iterators. We should probably add elements to the set one by one as they are produced by the underlying iterator.

Reply:
I hadn't yet fully considered infinite iterators yet. Thank you for catching this.

Reply:
The changes I just made "work" to consider each element at a time rather than traverse all elements into a set, but there is still a problem. `Iter[U32].repeat_value(7).unique()` will infinitely run as I still assume that eventually there will be either an end to the iterator or a new unique element. Any thoughts on the best way to prevent such infinite running?